File: 27.移除元素.java
/*
 * @lc app=leetcode.cn id=27 lang=java
 *
 * [27] 移除元素
 */

// @lc code=start
class Solution {
    public int removeElement(int[] nums, int val) {
        int first = 0;
        for(int i = 0; i < nums.length; i ++){
            if(nums[i] != val){
                nums[first ++] = nums[i];
            }
        }
        return first;
    }
}
// @lc code=end

File: 27.移除元素.go
/*
 * @lc app=leetcode.cn id=27 lang=golang
 *
 * [27] 移除元素
 */

// @lc code=start
func removeElement(nums []int, val int) int {
	first := -1
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			first = first + 1
			nums[first] = nums[i]
		}
	}
	return first + 1
}

// @lc code=end
